castai/resource_service_account.go

```diff
@@ -0,0 +1,175 @@
+package castai
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/castai/terraform-provider-castai/castai/sdk"
+)
+
+const (
+	FieldServiceAccountOrganizationID = "organization_id"
+	FieldServiceAccountName           = "name"
+	FieldServiceAccountID             = "service_account_id"
+	FieldServiceAccountDescription    = "description"
+	FieldServiceAccountEmail          = "email"
+
+	FieldServiceAccountAuthor      = "author"
+	FieldServiceAccountAuthorID    = "id"
+	FieldServiceAccountAuthorEmail = "email"
+	FieldServiceAccountAuthorKind  = "kind"
+)
+
+func resourceServiceAccount() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceServiceAccountCreate,
+		ReadContext:   resourceServiceAccountRead,
+		DeleteContext: resourceServiceAccountDelete,
+
+		Description: "Service Account resource allows managing CAST AI service accounts.",
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(3 * time.Minute),
+			Update: schema.DefaultTimeout(3 * time.Minute),
+			Delete: schema.DefaultTimeout(3 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			FieldServiceAccountOrganizationID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "ID of the organization.",
+			},
+			FieldServiceAccountName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the service account.",
+			},
+			FieldServiceAccountDescription: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
```

Comment:
Should the resource be recreated on Name or Description change?
We do not allow to update the Service Account - is it on purpose?

Reply:
We didn't create the endpoint for it as there was no need from the UI perspective

Reply:
But I'm gonna add the method as it doesn't make sense to recreate resources just because of name or description.

Reply:
Updated